v-next/hardhat-node-test-reporter/src/reporter.ts

```diff
@@ -149,21 +149,58 @@ export function hardhatTestReporter(
               }
             }
 
-            // If a suite/describe was already printed, we need to descrease
-            // the lastPrintedIndex, as we are removing it from the stack.
-            //
-            // If its nesting was 0, we print an empty line to separate top-level
-            // describes.
             if (event.data.nesting === 0) {
-              lastPrintedIndex = undefined;
+              // If last printed index is undefined it means that we haven't
+              // printed anything related to the top of the stack yet. If we
+              // would have, lastPrintedIndex would be 0 instead.
+              // This means that it's an empty or skipped describe.
+              if (lastPrintedIndex === undefined) {
+                const top = stack.slice(0, 1);
+
+                yield formatTestContext(top, "", "", event.data.skip === true);
+                yield "\n";
+              }
+
+              // If the describe nesting was 0, we print an empty line to separate
+              // top-level describes.
               yield "\n";
             } else {
-              if (lastPrintedIndex !== undefined) {
-                lastPrintedIndex = lastPrintedIndex - 1;
+              // If we are in this situation, we haven't printed anything
+              // related to the describe of the top of the stack yet, so it's
+              // an empty/skipped describe. Hence, we print it.
+              if (
+                lastPrintedIndex === undefined ||
+                lastPrintedIndex < stack.length - 1
+              ) {
+                const unprinted = stack.slice(
+                  lastPrintedIndex !== undefined ? lastPrintedIndex + 1 : 0,
+                );
```

Comment:
We could simplify this a little bit.

```suggestion
              if (lastPrintedIndex !== stack.length - 1) {
                const unprinted = stack[stack.length - 1];
```